cli/src/main/java/org/jboss/as/cli/parsing/ParserUtil.java

```diff
@@ -238,7 +238,7 @@ public void leavingState(ParsingContext ctx) throws CommandFormatException {
                         }
                     } else {
                         final String value = buffer.toString().trim();
-                        if (ch == '=') {
+                        if (ch == '=' && !ctx.isEndOfContent()) {
```

Comment:
Perhaps you could add a comment saying something like:
"if '=' is escaped, it will be not considered as a type/name separator. There is an exception when it is the last parsed character (e.g.: "test=foo\="). In this case, if we are at the end of the content, we don't consider it a separator. If we are not at the end of content, we consider it a separator."